frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -510,6 +498,37 @@ The interval value is in seconds and can range from 3 to 20 seconds.]]),
             end,
         })
     end
+    table.insert(menu_items.long_press.sub_item_table, {
+        text_func = function()
+            return T(_("Highlight long-press interval: %1 s"),
+                G_reader_settings:readSetting("highlight_long_hold_threshold_s", 3))
+        end,
+        keep_menu_open = true,
+        callback = function(touchmenu_instance)
+            local SpinWidget = require("ui/widget/spinwidget")
+            local items = SpinWidget:new{
+                title_text = _("Highlight long-press interval"),
+                info_text = _([[
+If a touch is not released in this interval, it is considered a long-press. On document text, single word selection will not be triggered.
+
+The interval value is in seconds and can range from 3 to 20 seconds.]]),
```

Comment:
This strikes me as an odd choice. The values of interest are roughly from 2-5 s with tenths or at the very least halves of seconds.

Reply:
Done: shows tenths, step is 0.5, any value can be entered via inputbox.

Reply:
Text hasn't changed yet ;-)

Btw, should the minimum really be as high as 3?

Reply:
I'm not opposed to decrease the minimum.
But I guess it should be larger ("significantly" larger ? meaning at least 0.5/1s larger, for a brain to compute the extra duration needed?) than the generic "Long press interval" in the Gesture interval submenu, which is (I think) the duration to trigger normal long-press (to get dict lookup).

Rereading the text and wondering if I'm not confusing things: I think I'm not. The text is confusing: it's not your regular "long-press on text" it should be describing, but the "long-long-press" interval, when a long-press behaves differently than your regular long-press.

The setting name is right, but the UI text is not :)
hold = long-press
long_hold = long-long-press :)

Reply:
I didn't say anything about the default value, but this hardly seems like a case where the default value should be the minimum. ^_^

I'd just make the minimum 1. A bit low perhaps, but more equivalent to 10 on the max than 20!

Reply:
Not minding that, we could have a hard minimum value of 1.
But we somehow need to enforce `highlight_long_hold_threshold_s` (that we configure here) to be larger than `ges_hold_interval_ms` which is configured elsewhere.
I have no idea what is the behaviour when the former is smaller than the latter.

Reply:
Do we want to actively prevent someone from saying they'd like this first and foremost?

Reply:
<img width="200" alt="72" src="https://user-images.githubusercontent.com/62179190/202703793-deae393d-887a-4d5a-b273-ae1e190d1da0.png">

Reply:
There is a checkbox `[x] Dictionary on single word selection` that thos who'd like this first and foremost can uncheck.
With _what is the behaviour when the former is smaller than the latter_, I'm saying this is untested (so, possibly undefined :) behaviour.

Reply:
<img width="200" alt="73" src="https://user-images.githubusercontent.com/62179190/202704228-a566f4a8-7eac-4dd9-be96-a001ae65ce81.png">

Reply:
But as shown on the above screenshot, if we enforce normal-long-press to be max 2s, if we also enforce very-long-press to be min 2s (or better, above 2.5s or 3s), we don't need to handle the dynamic interactions between the two :)

Reply:
Are milliseconds needed? It can be just seconds with one decimal digit, same as long-long-press.
Long-press: min 0.1, max 2.0, step 0.1, hold-step 0.5
Long-long-press: min 2.5, max 20.0, step 0.5, hold-step 1.0
Any number can be entered via the input field of the numberpicker.

Reply:
100ms units are ok, so fine with me.

Reply:
"The interval value is in seconds and can range from 3 to 20 seconds."
I think this text is redundant. We see the unit in the spinwidget, we see the min/max values in the input field when we tap the number.

Reply:
Indeed, the first part ("interval value is in seconds") was needed when we displayed just a number, which was in some places seconds, and milliseconds in others - so it's not needed nowadays.
The seconds part (range) gives some info more easily than having to launch the spin widget to get them - but that information is not really needed for the user at this point (although it gives a small hint about what we may think are good values, with the limits we have set).
So, ok for removing these, from all similar instances.